internal/provider/credential_providers_data_source.go

```diff
@@ -177,6 +177,22 @@ func (d *credentialProvidersDataSource) Schema(_ context.Context, _ datasource.S
 								"credential_style": schema.StringAttribute{
 									Computed: true,
 								},
+								"custom_parameters": schema.SetNestedAttribute{
+									Computed: true,
+									NestedObject: schema.NestedAttributeObject{
+										Attributes: map[string]schema.Attribute{
+											"key": schema.StringAttribute{
+												Computed: true,
+											},
+											"value": schema.StringAttribute{
+												Computed: true,
+											},
+											"value_type": schema.StringAttribute{
```

Comment:
We should have 2 additional bits for value_type:
1) We should default to `literal` so a customer doesn't have to provide it.
2) We should have validation for this field that requires it to be either `literal` or `dynamic`